pkg/provider/azure_storageaccount.go

```diff
@@ -160,6 +162,14 @@ func (az *Cloud) EnsureStorageAccount(ctx context.Context, accountOptions *Accou
 		subsID = accountOptions.SubscriptionID
 	}
 
+	if len(accountOptions.StorageEndpointSuffix) == 0 {
+		accountOptions.StorageEndpointSuffix = az.cloud.Environment.StorageEndpointSuffix
+	}
+
+	if accountOptions.CreatePrivateEndpoint {
+		privateDNSZoneName = fmt.Sprintf(privateDNSZoneName, accountOptions.StorageEndpointSuffix)
```

Comment:
don't use same var, otherwise `privateDNSZoneName` would be appened again and again